File: Python/problem2.py
def fibonacciUnderCap(cap, even):
    val = 1
    prevVal = 1
    total = 0
    while val <= cap:
        if(even and val%2 == 0):
            total = total + val
        newVal = val + prevVal
        prevVal = val
        val = newVal
    return total

print(fibonacciUnderCap(4000000, True))

File: C/problem2.c
#include <stdio.h>
#include <stdlib.h>

int fibonacciUnderCap(cap,even) {
    int curr, prev, next, total;
    curr = prev = 1;
    total = 0;
    while (curr <= cap) {
        if (even && curr%2 == 0)
            total = total + curr;
        next = curr + prev;
        prev = curr;
        curr = next;
    }
    return total;
}

int main(int argc, char * argv[]) {
    int True = 1;
    printf("%d\n", fibonacciUnderCap(4000000,True));
}
